packages/cli/src/UserManagement/auth/jwt.ts

```diff
@@ -2,39 +2,30 @@
 /* eslint-disable import/no-cycle */
 
 import * as jwt from 'jsonwebtoken';
+import { Response } from 'express';
+import { createHash } from 'crypto';
 import { JwtToken, PublicUserData } from '../Interfaces';
 import { User } from '../../databases/entities/User';
 import config = require('../../../config');
 
 export async function issueJWT(user: User): Promise<JwtToken> {
-	// Clean up null user data so it won't break during JWT generation
-	const optionalProperties = [
-		'email',
-		'firstName',
-		'lastName',
-		'password',
-		'personalizationAnswers',
-	];
-	optionalProperties.forEach((optionalProperty) => {
-		// @ts-ignore
-		if (!user[optionalProperty]) {
-			// @ts-ignore
-			// eslint-disable-next-line no-param-reassign
-			delete user[optionalProperty];
-		}
-	});
-	const { id, email, firstName, lastName, password, personalizationAnswers } = user;
+	const { id, email, password } = user;
 	const expiresIn = 14 * 86400000; // 14 days
 
 	const payload = {
 		id,
-		email,
-		firstName,
-		lastName,
-		password: password ? password.slice(Math.round(password.length / 2)) : undefined,
-		personalizationAnswers,
 	} as PublicUserData;
 
+	if (email) {
+		payload.email = email;
+	}
+
+	if (password) {
+		payload.password = createHash('sha256')
+			.update(password.slice(Math.round(password.length / 2)))
```

Comment:
If you want to have more than you should use `Math.floor` or even omit the rounding, since slice will round down and take the integer value of it example:

```js
var password = '12345678910'
password.slice(Math.ceil(password.length / 2)) --> '78910'
password.slice(Math.floor(password.length / 2)) --> '678910'
password.slice(password.length / 2) --> '678910'
```